src/main/java/de/dennisguse/opentracks/chart/ChartView.java

```diff
@@ -396,6 +418,31 @@ public boolean applyReportSpeed() {
         return false;
     }
 
+    void setShowElevation(boolean value) {
+        showElevation = value;
+        elevationSeries.setEnabled(value);
+    }
+    boolean getShowElevation() { return showElevation; }
+    void setShowPaceOrSpeed(boolean value) {
+        showPaceOrSpeed = value;
+
+        if (!showPaceOrSpeed) {
+            paceSeries.setEnabled(false);
+            speedSeries.setEnabled(false);
+            return;
+        }
+
+        // we want to make sure we show whatever version the user has
+        // selected when we turn this back on.
+        applyReportSpeed();
+    }
+    boolean getShowPaceOrSpeed() { return showPaceOrSpeed; }
+    void setShowHeartRate(boolean value) {
+        showHeartRate = value;
```

Comment:
You can use isEnabled() on each Series: no need to store this value here also.

Reply:
@dennisguse while this is true for `showHeartRate` and `showElevation`, the boolean tracking is necessary for `showPaceOrSpeed` because of the interaction between the two elements on the `Chart` itself. If you prefer, I can remove it for just the elevation and heart rate, but I could imagine someone returning and needing it again, if they work on this code further.